packages/framework-provider-aws/src/library/events-adapter.ts

```diff
@@ -110,3 +108,57 @@ export async function storeEvents(
   await dynamoDB.batchWrite(params).promise()
   logger.debug('[EventsAdapter#storeEvents] EventEnvelope stored')
 }
+
+export async function destroyEntity(
```

Comment:
I've included the pagination on deletions and the destroyEntity method inside events-adapter.ts is self contained. There is no need to execute two queries to get snapshots and then events, now we are performing a paginated scan, filtering by entityName and entityID getting 25 items per page. Also I've restored the readEntityEventsSince to its original version since I'm not using it.

In addition,in the integration tests I've included at least one hundred product updates to verify that all the events and snapshots are being deleted.